lib/active_model/serializer.rb

```diff
@@ -167,7 +167,7 @@ def serializable_hash(adapter_options = nil, options = {}, adapter_instance = se
       adapter_options ||= {}
       options[:include_directive] ||= ActiveModel::Serializer.include_directive_from_options(adapter_options)
       cached_attributes = adapter_options[:cached_attributes] ||= {}
```

Comment:
yeah, since if you're serializing a resource serializer  (not collection) and you don't have cached_attributes, you're not going to get them.  this particular attribute is added in the collection serializer. so, here we're just making sure it's set.